app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Resources\UserCollection;
+use App\Http\Resources\UserResource;
+use App\Models\Dictionary;
+use App\Models\DictionaryTerm;
+use App\Models\User;
+use DateTime;
+use Illuminate\Bus\Dispatcher;
+use Illuminate\Http\Request;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+
+class UserController extends Controller
+{
+
+    /**
+     * @OA\Get(
+     *      path="/users",
+     *      operationId="getUsersList",
+     *      tags={"Users"},
+     *      summary="Get list of users",
+     *      description="Returns list of users",
+     *      @OA\Response(
+     *          response=200,
+     *          description="Successful operation",
+     *          @OA\JsonContent(ref="#/components/schemas/UserResource")
+     *       ),
+     *      @OA\Response(
+     *          response=401,
+     *          description="Unauthenticated",
+     *      ),
+     *      @OA\Response(
+     *          response=403,
+     *          description="Forbidden"
+     *      )
+     *     )
+     */
+    public function index()
+    {
+        return new UserCollection(User::All());
+    }
+
+    /**
+     * @OA\Post(
+     *      path="/users",
+     *      operationId="storeUser",
+     *      tags={"Users"},
+     *      summary="Store new user",
+     *      description="Returns user data",
+     *      @OA\RequestBody(
+     *          required=true,
+     *          @OA\JsonContent(ref="#/components/schemas/StoreUserResource")
+     *      ),
+     *      @OA\Response(
+     *          response=201,
+     *          description="Successful operation",
+     *          @OA\JsonContent(ref="#/components/schemas/UserResource")
+     *       ),
+     *      @OA\Response(
+     *          response=400,
+     *          description="Bad Request"
+     *      ),
+     *      @OA\Response(
+     *          response=401,
+     *          description="Unauthenticated",
+     *      ),
+     *      @OA\Response(
+     *          response=403,
+     *          description="Forbidden"
+     *      )
+     * )
+     */
+    public function store(Request $request)
+    {
+        $user_role_id = 3;
```

Comment:
Tutaj można przenieść te id jeżeli zawsze będa stałe do constów w odpowiednim modelu będzie bardziej czytelne niż 3. Np. w DictionaryTerm.php dodać zmienną nazwa_roli_ID = 3 i tak dla każdej roli. Potem wołać z modelu te consty.

Reply:
Dodałem libkę https://github.com/BenSampo/laravel-enum, myślę, że to fajnie rozwiąże ten problem.

Reply:
wygląda spoko, zerknę w wolnym czasie na to